specs/experimental/security-council-vetoerset.md

```diff
@@ -0,0 +1,206 @@
+
+**Table of Contents**
+
+- [Context](#context)
+- [Implementation](#implementation)
+    - [Veto module](#veto-module)
+    - [Add owner module](#add-owner-module)
+    - [Owner guard](#owner-guard)
+    - [Adding owner call flow](#adding-owner-call-flow)
+    - [Performing a veto call flow](#performing-a-veto-call-flow)
+    - [Owner guard hooks call flow](#owner-guard-hooks-call-flow)
+- [Contracts interdependencies](#contracts-interdependencies)
+
+## Context
+
+The Security Council's veto mechanism is intended to allow any member of the VetoerSet to perform a veto across the entire Superchain.
+
+This veto mechanism is intended only to be used in abnormal cirumstances, such as a malicious chain upgrade, that would negatively impact or hurt a chain.
+
+## Implementation
+
+The veto mechanism relies on two Safe modules (`AddOwnerModule` and `VetoModule`) and one Safe guard (`OwnerGuard`) described thereafter.
+
+### Veto module
+
+The term "VetoerSet" is an abstract concept not explicitly present in the code. Any owner of the Safe Account is a potential "vetoer", so the "VetoerSet" is essentially just the list of owners of the Safe Account.
+
+This module implements the core (and simple) logic of allowing any member of the VetoerSet (i.e., any owner of the Safe Account) to call the `veto()` method on the registered `DelayedVetoable` contract on behalf of the Safe Account and without requiring a signature quorum.
+
+This module has the following minimal interface:
+
+```solidity
+interface VetoModule {
+    /// @notice Passthrough for any owner to execute a veto on the `DelayedVetoable` contract.
+    /// @dev Revert if not called by an owner of the Safe Account.
+    function veto() external returns (bool);
+}
+```
+
+### Add owner module
+
+Although new owners can already be added by directly calling `addOwnerWithThreshold()` on the Safe Account, it requires a signature quorum.
+
+The `AddOwnerModule` allows the OP Foundation (immutable set at deployment in `admin`) to add new owners without having to provide a signature quorum.
+
+This module has the following minimal interface:
+
+```solidity
+interface VetoModule {
+    /// @notice Add a new owner address.
+    /// @dev Revert if not called by the whitelisted `admin` address.
+    /// @param addr The owner address to add.
+    function addOwner(address addr) external;
+}
+```
+
+### Owner guard
+
+The `OwnerGuard` contract adds two additional requirements (invariants) to the Safe Account, which must hold true in any circumstance:
+1. The Safe Account cannot have more owners than the configured `maxOwnerCount`.
+2. The Safe Account `threshold` (minimum number of signatures necessary to perform an arbitrary action on behalf of the Safe Account) must always be exactly equal to 66% of the current number of owners.
+
+This guard has the following minimal interface:
+
+```solidity
+interface VetoModule {
```

Comment:
More appropriate name?
```suggestion
interface VetoerGuard {
```

Reply:
The `Module` part of the name is to make it clearer that this is in fact a [Safe Module](https://docs.safe.global/advanced/smart-account-modules).

Reply:
It's a module that inherits the `BaseGuard` interface (per the line below)?

Reply:
Oh, yes my bad it is a copy/paste issue the name of the interface should be `OwnerGuard` as mentioned in the paragraph right above it. Fixing it